syncopy/datatype/methods/selectdata.py

```diff
@@ -363,6 +365,31 @@ def selectdata(data,
     selectMethod.compute(data, out, parallel=kwargs.get("parallel"),
                          log_dict=log_dct)
 
+    # Handle selection of waveform for SpikeData objects
+    if type(data) == spy.SpikeData and data.waveform is not None:
+        if inplace:
+            spy.log("Inplace selection of SpikeData with waveform not supported for the waveform.", level="WARNING")
+        else:
+            fauxTrials = [data._preview_trial(trlno) for trlno in data.selection.trial_ids]
+            spikes_by_trial = [f.idx[0] for f in fauxTrials]
+            spike_idx = np.concatenate([np.array(x).ravel() for x in spikes_by_trial])
+
+            # Copy the proper subset of the waveform dataset to `out`, the new `SpikeData` object.
+            hdf5_file_in = data._get_backing_hdf5_file_handle()
+            hdf5_file_out = out._get_backing_hdf5_file_handle()
+
+            # Copy the waveform dataset into the new file, trial by trial to prevent memory issues.
+            ds = hdf5_file_out.create_dataset('waveform', shape=(len(spike_idx), *data.waveform.shape[1:]), dtype=data.waveform.dtype)
+            cur_new_idx = 0
+            for tidx, old_trial_indices in enumerate(spikes_by_trial):
+                num_spikes_this_trial = len(old_trial_indices)
+                #new_indices = range(cur_new_idx, cur_new_idx + num_spikes_this_trial)
+                new_indices = np.arange(cur_new_idx, cur_new_idx + num_spikes_this_trial)
```

Comment:
Here a `slice` could be good, not sure if h5py is so clever..

Reply:
ok just did an ad hoc `%timeit`, converting to slice would be faster, I will ad this with a commit